src/core/border_router/routing_manager.cpp

```diff
@@ -2792,16 +2792,21 @@ void RoutingManager::Nat64PrefixManager::GenerateLocalPrefix(const Ip6::Prefix &
 
 const Ip6::Prefix &RoutingManager::Nat64PrefixManager::GetFavoredPrefix(RoutePreference &aPreference) const
 {
-    const Ip6::Prefix *favoredPrefix = &mInfraIfPrefix;
+    const Ip6::Prefix *favoredPrefix = &mLocalPrefix;
+
+    aPreference = NetworkData::kRoutePreferenceLow;
 
     if (mInfraIfPrefix.IsValidNat64())
     {
-        aPreference = NetworkData::kRoutePreferenceMedium;
-    }
-    else
-    {
-        favoredPrefix = &mLocalPrefix;
-        aPreference   = NetworkData::kRoutePreferenceLow;
+        Ip6::Prefix     omrPrefix;
+        RoutePreference omrPrefixPreference = RoutePreference::kRoutePreferenceLow;
+
+        IgnoreError(Get<RoutingManager>().GetFavoredOmrPrefix(omrPrefix, omrPrefixPreference));
+        if (omrPrefixPreference == NetworkData::kRoutePreferenceMedium)
```

Comment:
Should we check for `>=` here?

Reply:
Actually, I think the check of preference here is to see if the OMR prefix is infrastructure-derived or not?

We can use `Get<RoutingManager>().mFavoredOmrPrefix.IsInfrastructureDerived()`  for this.
https://github.com/openthread/openthread/blob/fb45cf23fdfa710a154fa53554393f0f3b2bd3a0/src/core/border_router/routing_manager.cpp#L1902-L1909

I think we can simplify this block with the following?
```c++
    if (mInfraIfPrefix.IsValidNat64() && Get<RoutingManager>().mFavoredOmrPrefix.IsInfrastructureDerived())
    {
          favoredPrefix = &mInfraIfPrefix; 
          ...
```

Reply:
Updated. Thanks!